src/crimson/osd/osd_operations/client_request_common.cc

```diff
@@ -73,6 +73,12 @@ CommonClientRequest::do_recover_missing(
       !pg->is_degraded_or_backfilling_object(soid)) {
     return seastar::now();
   }
+  if (ver == eversion_t()) {
```

Comment:
nit, instead of checking, we can unconditionally set `ver` here and remove the previous user.

```
-  if (!pg->is_unreadable_object(soid, &ver) &&
+  if (!pg->is_unreadable_object(soid) &&```

Reply:
I'd somewhat prefer to avoid the branch and extra logic even at the expense of an extra std::map lookup.